### PR TITLE
Setup Google Oauth

### LINKED ISSUE #74: Set up authentication for NYPL staff

### PR DESCRIPTION
**Pull request by courte**
Allows folks with nypl.org emails to become admins. Sets up Google Oauth for authentication.

Fixes #74.

### CODE REVIEW

File: controller.py [hunk 1]
```diff
@@ -394,6 +394,53 @@ def apply_borrowing_policy(self, patron, license_pool):
...
+            test_mode=self.manager.testing
+        )
+
+    def authenticated_admin_from_request(self):
+        authorization = flask.request.environ.get('HTTP_AUTHORIZATION')
+        if authorization:
+            admin = get_one(self._db, Admin, access_token=authorization[7:])
+            if admin and self.google.active_credentials(admin):
+                return admin
+        return self.google.auth_uri
```
Comment (aslagle) [inline controller.py]: Is it possible to do the redirect here?

File: controller.py [hunk 2]
```diff
@@ -394,6 +394,53 @@ def apply_borrowing_policy(self, patron, license_pool):
...
+
+    def authenticated_admin(self, credentials):
+        """Creates or updates an admin with the given credentials"""
+        admin = Admin(email=credentials['email'])
+        admin.update_credentials(
+            self._db, credentials['access_token'], credentials['credentials']
```
Comment (aslagle) [inline controller.py]: what's in credentials["credentials"]?

Comment (aslagle) [inline controller.py]: oh I see, you've already extracted things from the google response.

File: controller.py (hunk 1 above)
Comment (courte) [inline controller.py]: There were some type issues, but I've found a less bad-regex-y way to return the redirect from here.

File: controller.py (hunk 2 above)
Comment (courte) [inline controller.py]: I also changed the naming here, since "credentials" wasn't accurate.

**Comment by courte** [comment]
Replaced by #86.

**Pull request closed without merging**